xbmc/pvr/PVRManager.cpp

```diff
@@ -594,12 +573,31 @@ bool CPVRManager::SetWakeupCommand(void)
 
 void CPVRManager::OnSleep()
 {
+  SetWakeupCommand();
+  SaveLastPlayedChannel();
+
   g_PVRClients->OnSystemSleep();
 }
 
 void CPVRManager::OnWake()
 {
   g_PVRClients->OnSystemWake();
+
+  if (IsStarted())
```

Comment:
I just spotted, that CPVRManager::Announce always() does an IsStarted check before doing anything.

Okay, but then you should add the IsStarted check also to CPVRManager::OnSleep() and CPVRManager::Deinit(), not only to CPVRManager::OnWake()

Finally, the IsStarted check should be removed from CPVRManager::TriggerSearchMissingChannelIcons().